### ISSUE TITLE
Support migration of project history

### ISSUE THREAD

**Issue by okorach-sonar** [sonar-solutions/sonar-reports#554]
**Disclaimer**: The would be an proof-of-concept for the moment, and it could potentially prove to not work

There could be a possibility for the migration tool to migrate project history. The high level principles of this concept would be:
- Leverage the undocumented `sonar.projectDate` scanner property that allows to back date the date of analysis
- For each project, iterate on:
   - Start from the first analysis in the project history on SQS
   - Create a synthetic analysis report with all the measures of the project (issues, files are not needed (or fake ones can be created to simply let the analysis report be accept by SQC) since they are only preserved for the last analysis of any given branch)
   - Back date the scanner report. The way to back date should be looked up in the scanner source code.
   - Iterate to next data point in the history

- In order to not migrate too many data points, and to not create too long migrations, we should limit the number of data points that we migrate and set a minimum interval between data points

[EDIT]
- All metrics that can be historized in SQC should be migrated. This includes (non exhaustive list) issue counts according to different criterias (issue counts by std experience severity (blocker issues, critical, major, minor, info), issue count by type (code smells, bugs, vulns) or sw quality (maintainability issues, reliability issues, security issues) and all other counts by type), accepted and false positive counts etc...), as well as all the other historized metrics
- It's ok if some metrics are not migrated because they are not computed in the analysis report but on the SQC side (eg probably coverage %, maybe duplication % and others). When history migration is chosen, the migration report should list in the limitations that some measures history are not migrated, citing the ones we are certain are not (coverage %, and others as we identify them)

**Comment by joshua-quek-sonarsource** [comment]
PoC implemented in #557.

**What it does** — follows the approach in the description: enumerate the source project's analyses (`api/project_analyses/search`), pull each selected point's project-level measures (`api/measures/search_history`), synthesize a minimal report per point (measures only, no issues), backdate it by stamping `Metadata.AnalysisDate` — the report-side equivalent of `sonar.projectDate` — and submit it before the regular current-snapshot import.

Opt-in, off by default:

| Setting | Default |
|---|---|
| `--migrate_history` | `false` |
| `--history_max_points` | `10` |
| `--history_min_interval_days` | `30` |

Selection drops the newest analysis, applies a greedy oldest-first min-interval filter, then evenly subsamples to `max_points` so the migrated set still spans the whole history range rather than clustering at the old end.

**Live-verified end to end** against a real SonarQube Server (2026.3.1) source and a SonarQube Cloud staging organization, using `okorach-oss_sonar-tools` (100 analyses, 2021→2026) with `--history_max_points 8 --history_min_interval_days 30`. All 8 selected points were accepted by the Compute Engine, and the target's Activity page now shows a real trend:

```
2022-01-29  v1.11    ncloc  6576
2023-07-09  v2.7.1   ncloc 10327
2024-05-17  3.0      ncloc 11057
2024-12-23  3.8      ncloc 13631
2025-06-02  3.14     ncloc 14170
2025-11-14  3.17     ncloc 14734
2026-05-09  3.20     ncloc 17341
2026-09-02  3.21     ncloc 17379   <- regular current-snapshot import
```

So the concept holds: backdated synthetic reports do land as genuinely-dated points in Activity, not as the submission date. That confirms @okorach's scanner-level finding also holds through this tool's own report-building and submission path, which is a different code path from the scanner CLI.

**Two limitations worth knowing, both found during live testing:**

1. **Target-side retention still applies.** A 2021 point (~5.2 years old) was accepted — its CE task reported `SUCCESS` — but SonarQube Cloud housekeeping then pruned it, so it never appeared in Activity. Nothing the migration can do; worth setting the bounding flags with the target's retention window in mind.
2. **Each historical entry carries one placeholder file.** A report with a lone `PROJECT` component holding a raw measure is rejected by the CE, so every point includes a single empty `__history_snapshot__.<ext>` component for the measures to attach to — which is the "fake ones can be created" escape hatch the description anticipated.

Other known PoC gaps (all in the PR description and `docs/TRANSFER.md`): main branch only, measures only, and not resume-safe.
